### PR TITLE
MAINTAINERS: Remove DT bindings area

### PR DESCRIPTION
**Pull request by decsny**
Now that the dts/bindings filter has been removed from Devicetree Binding area in #75016, this area serves no real purpose.

Move the include/dt-bindings files to their respective areas.

Fix some of the orphaned dts/bindings paths caused by #75016

Fix the imx_spc.h file being in it's own pm/ folder instead of power/ like the other power related headers.

Add any binding with "zephyr" in the name to the devicetree area.

### CODE REVIEW

File: MAINTAINERS.yml [hunk 1]
```diff
@@ -1198,6 +1201,8 @@ Release Notes:
     - drivers/dma/
     - tests/drivers/dma/
     - include/zephyr/drivers/dma/
+    - dts/bindings/dma/
+    - include/zephyr/dt-bindings/
```
Comment (stephanosio) [inline MAINTAINERS.yml:1205]:
```suggestion
    - include/zephyr/dt-bindings/dma/
```

**Review by stephanosio — DISMISSED** [review]

File: MAINTAINERS.yml (hunk 1 above)
Comment (decsny) [reply MAINTAINERS.yml:1205]: thanks for the catch

File: dts/arm/nxp/nxp_rt11xx.dtsi [hunk 2]
```diff
@@ -11,7 +11,7 @@
 #include <zephyr/dt-bindings/gpio/gpio.h>
 #include <zephyr/dt-bindings/i2c/i2c.h>
 #include <zephyr/dt-bindings/pwm/pwm.h>
-#include <zephyr/dt-bindings/pm/imx_spc.h>
```
Comment (dleach02) [inline dts/arm/nxp/nxp_rt11xx.dtsi:14]: How is this related to the changes in maintainers? Is this a bug you are trying to fix?

File: soc/nxp/imxrt/imxrt11xx/power.c [hunk 3]
```diff
@@ -8,7 +8,7 @@
 #include <zephyr/pm/pm.h>
 #include <fsl_dcdc.h>
 #include <fsl_gpc.h>
-#include <zephyr/dt-bindings/pm/imx_spc.h>
```
Comment (dleach02) [inline soc/nxp/imxrt/imxrt11xx/power.c:11]: ditto comment

File: dts/arm/nxp/nxp_rt11xx.dtsi (hunk 2 above)
Comment (decsny) [reply dts/arm/nxp/nxp_rt11xx.dtsi:14]: fixing the path

File: soc/nxp/imxrt/imxrt11xx/power.c (hunk 3 above)
Comment (decsny) [reply soc/nxp/imxrt/imxrt11xx/power.c:11]: fixing the path

File: dts/arm/nxp/nxp_rt11xx.dtsi (hunk 2 above)
Comment (decsny) [reply dts/arm/nxp/nxp_rt11xx.dtsi:14]: when fixing the maintainer path of include/dt-bindings/power , I noticed there is this one single file that created its own folder called include/dt-bindings/pm which is really redundant, so I am fixing this file path

**Review by MaureenHelm — DISMISSED** [review]

File: MAINTAINERS.yml [hunk 4]
```diff
@@ -809,6 +811,8 @@ Devicetree:
   collaborators:
     - decsny
     - galak
+  files-regex:
+    - dts/bindings/.*zephyr.*
```
Comment (henrikbrixandersen) [inline MAINTAINERS.yml:815]: Not sure why any binding with zephyr in it should belong to this area?

Comment (decsny) [reply MAINTAINERS.yml:815]: why wouldn't they? those types of bindings should be reviewed by devicetree maintainers, as they should not be added lightly

Comment (decsny) [reply MAINTAINERS.yml:815]: in the past all the DT bindings were reviewed by DT maintainers but now with the recent change I dont get added to review any bindings, and apparently now we have to rely on the review of the other area maintainers to review the DT usage properly, and honestly, I cannot say that I trust all the other maintainer of different areas to have the same type of glasses on when reviewing DT bindings. And what makes this more annoying is that nobody consulted me about this change to this DT binding area even though I am the only active one on the list (did somebody ask Kumar before making this change at least?) Regardless I am exhausted already from reviewing so many PRs due to that filter, so I am okay to limit it just to ones that have `zephyr` in the name

**Review by dleach02 — DISMISSED** [review]

**Comment by fabiobaltieri** [comment]
hey, this needs a rebase

**Review by MaureenHelm — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]
